### PR TITLE
[ONNX] Add repeat_interleave symbolic

### PR DESCRIPTION
**Pull request by shubhambhokare1**
- Add support for aten::repeat_interleave
- NOTE: Also adds fix for cases with split op where input tensor sizes are not known but _outputs is provided

### CODE REVIEW

**Comment by BowenBao** [comment]
please check ci failure

File: torch/onnx/symbolic_opset9.py [hunk 1]
```diff
@@ -605,7 +605,10 @@ def split(g, self, split_size_or_sizes, dim, _outputs=None):
 
     size = sym_help._get_tensor_dim_size(self, dim)
     if size is None:
-        return sym_help._onnx_opset_unsupported_detailed('split', 9, 11, 'Unknown dimension size not supported')
+        if _outputs is not None:
+            size = _outputs
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset9.py:609]: this only works when split_size is 1

File: torch/onnx/symbolic_opset9.py [hunk 2]
```diff
@@ -1951,6 +1954,57 @@ def repeat(g, self, repeats):
...
+    for idx, input_size in enumerate(input_sizes):
+        if input_size is None:
+            input_sizes[idx], input_sizes_temp[idx] = 0, -1
+
+    # Cases where repeats is an int or single value tensor
+    if (repeats_dim == 0 or (repeats_dim == 1 and repeats_sizes[0] == 1)):
+        if not sym_help._is_tensor(repeats):
+            repeats = g.op("Constant", value_t=torch.LongTensor(repeats))
+        if input_sizes[dim] == 0:
+            raise RuntimeError("Input size along dimension to repeat must exist")
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset9.py:1980]:
```suggestion
            raise RuntimeError("Input size along dimension to repeat must be known at export time.")
```

File: torch/onnx/symbolic_opset9.py [hunk 3]
```diff
@@ -1951,6 +1954,57 @@ def repeat(g, self, repeats):
...
+    input = self
+    # if dim is None flatten
+    # By default, use the flattened input array, and return a flat output array
+    if sym_help._is_none(dim):
+        input = reshape(g, self, g.op("Constant", value_t=torch.tensor([-1])))
+        dim = 0
+    else:
+        dim = sym_help._maybe_get_scalar(dim)
+
+    repeats_dim = sym_help._get_tensor_rank(repeats)
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset9.py:1967]: These all could return None, please handle these cases.

File: torch/onnx/symbolic_opset9.py (hunk 2 above)
Comment (BowenBao) [inline torch/onnx/symbolic_opset9.py:1980]: i wonder if this is actually do-able, by using `_size_helper`, getting dim size through operator Shape and Gather dynamically.

File: torch/onnx/symbolic_helper.py [hunk 4]
```diff
@@ -598,6 +598,15 @@ def _scatter_helper(g, self, dim, index, src):
...
     return scatter(g, self, dim, index, src)
 
+def _repeat_interleave_split_helper(g, self, reps, dim):
+    if _export_onnx_opset_version <= 10:
+        return g.op("Split", self, split_i=[1] * reps, axis_i=dim, outputs=reps)
+    else:
+        from torch.onnx.symbolic_opset11 import split  # type: ignore
+        from torch.onnx.symbolic_opset9 import expand
+        repeats = g.op("Constant", value_t=torch.tensor([reps]))
+        repeats = expand(g, g.op("Constant", value_t=torch.tensor([1])), repeats, None)
```
[the remark this answers is not in the document]
Comment (shubhambhokare1) [reply torch/onnx/symbolic_helper.py:608]: the expand is to handle change in ONNX spec for split op in opset 13

File: torch/onnx/symbolic_opset9.py (hunk 1 above)
Comment (shubhambhokare1) [reply torch/onnx/symbolic_opset9.py:609]: Modified logic

File: torch/onnx/symbolic_opset9.py (hunk 3 above)
Comment (shubhambhokare1) [reply torch/onnx/symbolic_opset9.py:1967]: Added checks

File: torch/onnx/symbolic_helper.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (shubhambhokare1) [reply torch/onnx/symbolic_helper.py]: Removed expand node. Thanks

**Review by BowenBao — APPROVED** [review]
